server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/DefaultEventModule.java

```diff
@@ -67,25 +75,54 @@ protected void configure() {
     public static class ListenerRegistrationPerformer implements ConfigurationPerformer {
         private final MailboxManager mailboxManager;
         private final Set<MailboxListener> listeners;
+        private final PropertiesProvider propertiesProvider;
+        private final GuiceMailboxListenerLoader listenerLoader;
 
         @Inject
         public ListenerRegistrationPerformer(@Named(Names.MAILBOXMANAGER_NAME) MailboxManager mailboxManager,
-                                             Set<MailboxListener> listeners) {
+                                             Set<MailboxListener> listeners, PropertiesProvider propertiesProvider,
+                                             GuiceMailboxListenerLoader listenerLoader) {
             this.mailboxManager = mailboxManager;
             this.listeners = listeners;
+            this.propertiesProvider = propertiesProvider;
+            this.listenerLoader = listenerLoader;
         }
 
         @Override
         public void initModule() {
             try {
                 MailboxSession systemSession = mailboxManager.createSystemSession("storeMailboxManager");
-                listeners.forEach(Throwing.consumer(listener ->
-                    mailboxManager.addGlobalListener(listener, systemSession)));
-            } catch (MailboxException e) {
-                Throwables.propagate(e);
+
+                loadGuiceDefinedListeners(systemSession);
+                loadUserDefinedMailboxListeners(systemSession);
+            } catch (Exception e) {
+                throw Throwables.propagate(e);
             }
         }
 
+        public void loadUserDefinedMailboxListeners(MailboxSession systemSession) throws ConfigurationException {
+            try {
+                PropertiesConfiguration mailboxListeners = propertiesProvider.getConfiguration("mailboxListeners");
+
+                List<String> listenerNames = Splitter.on(',')
+                    .trimResults()
+                    .omitEmptyStrings()
+                    .splitToList(mailboxListeners.getString("listener.names", ""));
+
+                listenerNames.stream()
+                    .map(name -> mailboxListeners.getString(name + ".class"))
+                    .map(Throwing.function(listenerLoader::getListener))
```

Comment:
I'm not against a `.peek(LOGGER.info...` before `getListener` call, in order to know which classes are loaded.

Reply:
(this may also help us in case of failure)